### PR TITLE
Websockets API support

### PR DESCRIPTION
**Pull request by dynbit**
- Add VLTWsApiClient
- Start using protocol buffers from google

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.PHONY: default
+
+default:
+	echo "All your base are belong to us."
```
Comment (Antanas) [inline Makefile]: not sure if this is the best message

File: Tests/Core/VLTApiWsClientTests.m [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    VLTPBHandshakeResponse *response = [[VLTPBHandshakeResponse alloc] init];
+    response.sampleSize = 5;
+    response.captureInterval = 10;
+    response.canDetectMotion = YES;
+    response.canLabelMotion = YES;
+    [self.wsApiClient webSocket:self.mockWebSocket didReceiveMessage:[response data]];
+
+
+
```
Comment (Antanas) [inline Tests/Core/VLTApiWsClientTests.m]: too many spaces

File: VelocitySDK/Core/WebSockets/VLTWsApiClient.m [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        _ws = [[SRWebSocket alloc] initWithURL:url];
+        _ws.delegate = self;
+    }
+    return self;
+}
+
+- (void)webSocket:(SRWebSocket *)webSocket didReceiveMessage:(id)message
+{
+    if (self.queue.count > 0) {
+        VLTWsApiClientRequest *req = [self.queue get];
```
Comment (Antanas) [inline VelocitySDK/Core/WebSockets/VLTWsApiClient.m:54]: What will happen if:

```
...
[client sendData:data1 error:error];
[client sendData:data2 success:success failure:failure];
...
```

this delegate method will be called and success block will be invoked for request which didn't have one.

Maybe add ```getById:``` to queue ?

File: VelocitySDK/Core/WebSockets/VLTWsApiClient.m [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    VLTWsApiClientRequest *req = [[VLTWsApiClientRequest alloc] initWithData:data
+                                                                     success:success
+                                                                     failure:failure];
+    [self.queue add:req];
+    NSError *error = nil;
+    [self.ws sendData:data error:&error];
+    if (error != nil) {
+        [self webSocket:self.ws didFailWithError:error];
+        DLog(@"VLTApiWsClient send data error: %@", error);
+        return;
```
Comment (Antanas) [inline VelocitySDK/Core/WebSockets/VLTWsApiClient.m]: redundant

File: VelocitySDK/Core/WebSockets/VLTWsApiClient.m [hunk 5]
```diff
@@ -0,0 +1,151 @@
+//
+//  VLTWsApiClient.m
+//  VelocitySDK iOS
+//
+//  Created by Vytautas Galaunia on 20/10/2017.
+//  Copyright © 2017 Veloctity. All rights reserved.
+//
+
+#import "VLTWsApiClient.h"
+#import "ListQueue.h"
```
Comment (Antanas) [inline VelocitySDK/Core/WebSockets/VLTWsApiClient.m]: this import not needed as there is ```FifoListQueue.h```

File: VelocitySDK/ListQueues/FifoListQueue.m [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+//
+
+#import "FifoListQueue.h"
+
+@interface FifoListQueue ()
+
+@property (nonatomic, strong) dispatch_queue_t dispatchQueue;
+@property (nonatomic, strong) NSMutableArray *storage;
+
+
```
Comment (Antanas) [inline VelocitySDK/ListQueues/FifoListQueue.m]: spacing (and two lines after implementation)

File: VelocitySDK/Core/WebSockets/VLTWsApiClient.m (hunk 3 above)
Comment (dynbit) [reply VelocitySDK/Core/WebSockets/VLTWsApiClient.m:54]:
```
[client sendData:data1 error:error];
```
Is for a message for which we don't expect to receive a response. 
So, basically nothing will happen.

**Pull request merged**